### DISCUSSION TITLE
Using vite-plugin-deno on Nuxt 3 with Deno as package manager

### DISCUSSION THREAD

**Discussion by DMark05** [deno-plc/vite-plugin-deno#4] [Q&A]
I'm trying to use Nuxt 3 and Deno for a small project I'm working on, but I'm having problems with imports, so I wanted to try using this Vite Plugin, since Nuxt uses Vite to resolve imports.
The problem I'm having right now is that I need to tell Nuxt to use the vite plugin, which is easy enough, as shown here: **[Using Vite Plugins in Nuxt](https://nuxt.com/docs/guide/recipes/vite-plugin)**, but now I need to import the plugin which is a deno plugin, so I get back to the original problem...
I'd also like to add that I don't have much experience with web development, so excuse me if it's an easy solution.

**Comment by hansSchall** [comment]
The problem is the configuration file `nuxt.config.ts`. Like `vite.config.ts` it is executed using Node.js, but this plugin uses Deno APIs...

I am currently not aware of any way to work around this, maybe Nuxt offers a JS API that can be used like the Vite API. In that case you could do the same as with Vite build scripts.

**Reply by DMark05** [reply]
nuxt.config.ts allows me to add Vite plugins, the only problem is I need to import the plugin. I tested manually moving a package from a project that was using npm and it worked on the Deno version. Is there any way to export the plugin in a way that I can just drop inside node_modules and tell nuxt to import from there?

**Reply by hansSchall** [reply]
The problem is that this plugin depends on Deno-only APIs, which are not implemented by Node.js. As far as I understand Nuxt, nuxt.config.ts is executed with Node.js. Even if you managed to import it, it wont work...

You can quickly test the support by adding `console.log(Deno.build)` to the config. In case you get an error like `Deno is undefined` this plugin wont work, if it displays build info install this plugin using `npx jsr add @deno-plc/vite-plugin-deno`

**Reply by DMark05** [reply]
When running `console.log(Deno build)`, I get the output `{ deno: "2.1.3", v8: "13.0.245.12-rusty", typescript: "5.6.2" }` and I was able to add the plugin using `npx jsr add @deno-plc/vite-plugin-deno` and I had to add a couple more dependencies manually (zod, std__path, resolve.exports, validate-npm-package-name),
Now the output is the following:
```
error: Uncaught (in worker "") TypeScript files are not supported in npm packages: file:///home/neohercules/projects/projeto_inov/nuxt-deno/node_modules/@deno-plc/vite-plugin-deno/src/ast-worker.ts
 ERROR  [unhandledRejection] Unhandled error in child worker.                                                                                            10:24:26 AM

  at Worker.#pollControl (ext:runtime/11_workers.js:204:19)
  at Object.runMicrotasks (ext:core/01_core.js:683:26)
  at Array.processTicksAndRejections (ext:deno_node/_next_tick.ts:59:10)
  at eventLoopTick (ext:core/01_core.js:172:29)

```
The unhandled error in child worker repeats for a bunch of different lines and the typescript one also shows up a bunch of times, but always looks the same.
If it is too troublesome for now, I may just use vite+vue instead of Nuxt, but tell me if you think there is anything I can do about this.

**Reply by hansSchall** [reply]
Looks like it is actually possible to bring this to work. I will investigate this, but it might take some time.
